packages/discovery-react-components/src/utils/useDataApi.ts

```diff
@@ -162,6 +160,53 @@ interface ReducerState {
   parameters: any;
 }
 
+/**
+ * concrete implementation of the reducer state for fetchFields
+ */
+export interface FieldsStore extends ReducerState {
+  data: DiscoveryV2.ListFieldsResponse | null;
+  parameters: DiscoveryV2.ListFieldsParams;
+}
+
+/**
+ * actions used to interact with the fields api and fields state
+ */
+export interface FieldsStoreActions {
+  /**
+   * method used to invoke the search request with an optional callback to return the response data
+   */
+  fetchFields: (callback?: (results: DiscoveryV2.ListFieldsResponse) => void) => void;
+  setFieldsResponse: (overrideResults: DiscoveryV2.ListFieldsResponse | null) => void;
+}
+
+/**
+ * concrete usage of the useDataApi helper method for fetching project fields
+ * @param searchParameters - initial search parameters to set
+ * @param searchClient - search client used to perform requests
+ * @return a 2-element array containing the fields store data and fields-specific store actions
+ */
+
+export const useFieldsApi = (
+  fetchFieldsParams: DiscoveryV2.ListFieldsParams,
+  searchClient: SearchClient
+): [FieldsStore, FieldsStoreActions] => {
+  const { state: filedsState, setData: setFieldsResponse, setFetchToken } = useDataApi(
+    fetchFieldsParams,
+    null,
+    searchClient.listFields,
+    searchClient
+  );
+  return [
+    filedsState,
+    {
+      fetchFields: (callback?: (results: DiscoveryV2.ListFieldsResponse) => void) => {
```

Comment:
i don't think you are using the `callback` here, so its just added complexity that can be removed. i think this can just be
```suggestion
      fetchFields: () => {
        setFetchToken({ trigger: true })
      }
```